src/screens/TopicPageScreen/elements/BottomSheetFollow.tsx

```diff
@@ -0,0 +1,307 @@
+          getTopicDetail(topicName);
+        } catch (error) {
+          if (__DEV__) {
+            console.log(error);
+          }
+        }
+      }
+    }
+  };
+
+  const handleUnfollowTopic = async ({type}: {type: Follow}) => {
+    setFollowType('');
+    setIsFollow(false);
+    setMemberCount(memberCount - 1);
+    onClose();
+    try {
+      const isIncognito = type === 'incognito';
+      await followTopic(topicName, isIncognito);
+    } catch (error) {
+      if (__DEV__) {
+        console.log(error);
+      }
+    }
+  };
+
+  return (
+    <View>
+      <BottomSheetV2
+        ref={ref}
+        closeOnPressMask={true}
+        height={290}
+        keyboardAvoidingViewEnabled={true}
+        viewstyle={{
+          paddingHorizontal: 0,
+          paddingTop: 0
+        }}>
+        <View style={styles.containerBottomSheet}>
+          <View style={styles.titleContainer}>
+            <View
+              style={{
+                width: 12,
+                height: 12
+              }}
+            />
+            <Text style={styles.title}>Join as</Text>
+
+            <TouchableOpacity onPress={onClose}>
+              <IconClose color={COLORS.black} />
+            </TouchableOpacity>
+          </View>
+          <View
+            style={{
+              borderTopColor: COLORS.gray,
+              borderTopWidth: 1
+            }}
+          />
+          <UserItem
+            type="signed"
+            username={props.username}
+            profilePicture={props.profilePicture}
+            isFollowing={followType === 'signed'}
+            onPress={() => {
+              handleFollowTopic({
+                type: 'signed'
+              });
+            }}
+          />
+          <UserItem
+            type="incognito"
+            username="Incognito"
+            profilePicture=""
+            isFollowing={followType === 'incognito'}
+            onPress={() => {
+              handleFollowTopic({
+                type: 'incognito'
+              });
+            }}
+          />
+          <View
+            style={{
+              paddingVertical: dimen.normalizeDimen(14),
+              paddingHorizontal: dimen.normalizeDimen(20)
+            }}>
+            <Button
+              onPress={() => {
+                handleUnfollowTopic({
+                  type: followType
+                });
+              }}
+              disabled={followType === ''}
+              styles={{
+                backgroundColor: COLORS.white,
+                paddingVertical: dimen.normalizeDimen(11),
+                paddingHorizontal: dimen.normalizeDimen(20),
+                borderRadius: dimen.normalizeDimen(8),
+                height: dimen.normalizeDimen(40),
+                flexDirection: 'row',
+                justifyContent: 'center',
+                alignItems: 'center',
+                borderColor: isFollow ? COLORS.redalert : COLORS.light_silver,
+                borderWidth: 1
+              }}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The styling for the "Leave Community" button is hardcoded within the component. This can make it difficult to reuse this component with different styles in the future. Consider moving these styles to the StyleSheet at the bottom of the file or creating a new component for this button if it's used in multiple places.

```diff
-              styles={{
-                backgroundColor: COLORS.white,
-                paddingVertical: dimen.normalizeDimen(11),
-                paddingHorizontal: dimen.normalizeDimen(20),
-                borderRadius: dimen.normalizeDimen(8),
-                height: dimen.normalizeDimen(40),
-                flexDirection: 'row',
-                justifyContent: 'center',
-                alignItems: 'center',
-                borderColor: isFollow ? COLORS.redalert : COLORS.light_silver,
-                borderWidth: 1
-              }}
```

Overall, the code looks good but could benefit from some refactoring for better readability and maintainability.


<!-- This is an auto-generated comment by OSS CodeRabbit -->